File: Category/Hash Map/0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int , int> map;
        for (int i = 0 ; i< nums.size() ; ++i) {
            int n = nums[i];
            if(map.find(n) != map.end()) {
                return { map[n], i};
            }
            int diff = target - nums[i];
            map[diff] = i;
        }
        return {-1, -1};
    }
};

File: Problems/0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}
        for i, n in enumerate(nums):      
            if n in hashMap:
                return [i, hashMap[n]]
            diff = target - n
            hashMap[diff] = i